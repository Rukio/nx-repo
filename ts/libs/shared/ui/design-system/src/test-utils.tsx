import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider, theme } from './lib';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import userEvent from '@testing-library/user-event';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'wrapper'>;
  userEventOptions?: UserEventSetupConfig;
};

const WrappedProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement,
  options?: CustomRenderOptions
) => {
  return {
    user: userEvent.setup(options?.userEventOptions),
    ...render(ui, { wrapper: WrappedProviders, ...options?.renderOptions }),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
