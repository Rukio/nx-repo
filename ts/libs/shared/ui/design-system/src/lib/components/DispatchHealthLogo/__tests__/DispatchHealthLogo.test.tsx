import { render } from '../../../../test-utils';

import *company-data-covered*Logo from '../index';

test('should render default size', () => {
  const { asFragment } = render(<*company-data-covered*Logo />);
  expect(asFragment()).toMatchSnapshot();
});

test('should render custom size', () => {
  const { asFragment } = render(<*company-data-covered*Logo pixelHeight={48} />);
  expect(asFragment()).toMatchSnapshot();
});
