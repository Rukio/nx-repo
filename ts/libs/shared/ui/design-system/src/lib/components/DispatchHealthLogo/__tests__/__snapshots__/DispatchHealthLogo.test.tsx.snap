// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render custom size 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="48"
    viewBox="0 0 348 42"
    width="408"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      clip-path="url(#clip0_2_2304)"
    >
      <g
        clip-path="url(#clip1_2_2304)"
      >
        <path
          d="M25.7069 1.06855V26.2614H28.2776V32.8081H18.6804V30.2716C17.138 32.2596 14.9786 33.2879 12.0651 33.2879C5.89546 33.2536 0 27.8723 0 20.6401C0 13.6135 5.75835 8.33505 11.8937 8.33505C14.4987 8.33505 16.7952 9.29477 18.6804 11.0771V7.58098H16.1097V1.06855H25.7069ZM18.886 20.6401C18.886 17.4867 16.3153 14.9846 13.0591 14.9846C9.80291 14.9846 7.30077 17.6924 7.30077 20.8115C7.30077 23.9991 9.97429 26.5698 13.0591 26.5698C16.3839 26.6041 18.886 23.7935 18.886 20.6401Z"
          fill="#EF3200"
        />
        <path
          d="M40.2057 8.84919V26.2614H42.7078V32.8081H30.6084V26.2614H33.1448V15.3273H30.6084V8.84919H40.2057ZM40.2057 7.58098H33.1448V1.06855H40.2057V7.58098Z"
          fill="#EF3200"
        />
        <path
          d="M58.2691 8.74636H62.9306V15.6701H56.5553C56.4182 14.2991 55.5613 13.545 54.156 13.545C52.8535 13.545 51.8937 14.2648 51.8937 15.2931C51.8937 16.6298 53.2991 17.2125 56.7267 18.0351C59.7772 18.7892 61.6967 19.7147 62.862 20.9143C64.0274 22.1482 64.6101 23.7249 64.6101 25.6101C64.6101 29.9974 61.1482 33.2879 55.9383 33.2879C53.3676 33.2879 51.5853 32.7052 49.8372 31.2314V32.7395H45.0043V25.8843H51.7566C52.0309 27.5296 52.9563 28.2494 54.7386 28.2494C56.3496 28.2494 57.138 27.3925 57.138 26.3299C57.138 25.1302 56.2811 24.5133 53.4362 23.7249L51.2082 23.108C49.8372 22.7652 48.7746 22.3539 47.7806 21.7369C45.8269 20.503 44.4216 18.6178 44.4216 15.8758C44.4216 11.3513 48.2262 8.33505 53.1277 8.33505C55.3556 8.33505 56.898 8.81491 58.2691 10.2545V8.74636Z"
          fill="#EF3200"
        />
        <path
          d="M81.988 33.2536C79.7601 33.2536 77.4636 32.7052 74.8929 30.5116V34.5219H77.4293V41H65.2956V34.5219H67.832V15.3273H65.2956V8.84919H74.8929V11.6255C76.4353 9.46615 78.6975 8.33505 81.5424 8.33505C88.0206 8.33505 93.6761 13.8877 93.6761 20.9143C93.6418 27.461 88.3976 33.2536 81.988 33.2536ZM86.4439 20.7772C86.4439 17.4524 83.7361 15.0189 80.4799 15.0189C77.2236 15.0189 74.6187 17.6924 74.6187 20.8458C74.6187 24.2734 77.4636 26.6041 80.5141 26.6041C83.8389 26.6041 86.4439 23.9649 86.4439 20.7772Z"
          fill="#EF3200"
        />
        <path
          d="M113.076 8.84919H122.674V15.3273H120.137V26.2614H122.674V32.8081H113.076V30.7515C111.534 32.5681 109.272 33.5621 106.495 33.5621C99.7429 33.5621 94.4644 27.5981 94.4644 20.9486C94.4644 13.922 100.017 8.09512 106.769 8.09512C109.443 8.09512 111.465 9.08912 113.076 11.1457V8.84919ZM113.111 20.7429C113.111 17.5895 110.608 15.0189 107.386 15.0189C104.165 15.0189 101.628 17.7266 101.628 20.88C101.628 24.1705 104.302 26.6041 107.386 26.6041C110.643 26.6041 113.111 23.8963 113.111 20.7429Z"
          fill="#EF3200"
        />
        <path
          d="M135.116 1.06855V8.84919H138.406V15.3273H135.116V26.2614H138.543V32.7738H128.055V15.3273H124.319V8.84919H128.055V1.06855H135.116Z"
          fill="#EF3200"
        />
        <path
          d="M157.224 8.84919H163.188V17.4524H156.607C155.407 15.9443 154.036 15.0189 152.048 15.0189C148.86 15.0189 146.632 17.7267 146.632 20.8458C146.632 24.102 149.066 26.6041 152.288 26.6041C154.036 26.6041 155.544 25.85 156.367 24.3762H164.182C162.571 29.7918 157.601 33.5279 152.048 33.5279C144.987 33.5279 139.469 27.6324 139.469 20.7772C139.469 13.8535 144.919 8.06084 151.465 8.06084C153.625 8.06084 155.51 8.81491 157.224 10.5973V8.84919Z"
          fill="#EF3200"
        />
        <path
          d="M174.327 1.06855V12.0026C175.938 9.56898 178.098 8.36932 180.908 8.36932C186.187 8.36932 189.443 12.8595 189.443 18.4807V26.2614H191.945V32.8081H182.348V20.1602C182.348 16.6641 181.045 15.2588 178.44 15.2588C175.733 15.2588 174.293 17.0411 174.293 20.0231V26.2271H176.829V32.7738H164.696V26.2614H167.232V7.58098H164.696V1.06855H174.327Z"
          fill="#EF3200"
        />
        <path
          d="M203.256 1.10283V13.2365C205.039 10.3573 207.369 8.74636 210.626 8.74636C213.093 8.74636 215.356 9.70608 217.001 11.6255C218.646 13.545 219.469 16.15 219.469 19.3719V28.7635H222.931V32.7395H211.757V28.7635H215.184V20.0231C215.184 17.5895 214.876 16.8355 214.396 15.91C213.368 14.0934 211.551 13.0308 209.357 13.0308C205.93 13.0308 203.29 15.3273 203.29 20.0917V28.7635H206.752V32.7395H195.578V28.7635H199.006V5.07883H195.578V1.10283H203.256Z"
          fill="#748491"
        />
        <path
          d="M247.404 22.5938H227.318C228.175 26.8783 231.397 29.5518 235.476 29.5518C238.355 29.5518 240.48 28.3865 242.125 25.9529H246.615C244.456 30.7515 240.24 33.3907 235.476 33.3907C228.723 33.3907 223.102 27.9752 223.102 21.0514C223.102 13.9563 228.689 8.60925 235.304 8.60925C241.851 8.60925 247.541 13.7164 247.541 21.12C247.541 21.6684 247.472 22.2853 247.404 22.5938ZM243.599 19.2691C242.674 15.156 239.314 12.4482 235.27 12.4482C231.74 12.4482 228.278 15.0189 227.318 19.2691H243.599Z"
          fill="#748491"
        />
        <path
          d="M270.231 9.2605H277.943V13.2365H274.584V28.7635H277.943V32.7395H270.231V28.7635C267.935 31.9512 265.261 33.2536 261.731 33.2536C255.081 33.2536 249.803 27.7009 249.803 21.1542C249.803 14.539 255.116 8.78063 261.902 8.78063C265.433 8.78063 268.003 10.1517 270.231 13.3393V9.2605ZM270.231 20.9143C270.231 16.6298 266.735 13.0308 262.108 13.0308C257.618 13.0308 254.122 16.6984 254.122 21.0514C254.122 25.4387 257.652 29.0034 262.108 29.0034C266.564 28.9692 270.231 25.4387 270.231 20.9143Z"
          fill="#748491"
        />
        <path
          d="M289.049 1.10283V28.7635H292.476V32.7395H281.097V28.7635H284.73V5.07883H281.268V1.10283H289.049Z"
          fill="#748491"
        />
        <path
          d="M300.6 1.10283V9.2605H305.124V13.2365H300.6V28.7635H305.433V32.7395H296.315V13.2365H292.236V9.2605H296.315V1.10283H300.6Z"
          fill="#748491"
        />
        <path
          d="M316.572 1.10283V13.2365C318.355 10.3573 320.685 8.74636 323.942 8.74636C326.41 8.74636 328.672 9.70608 330.317 11.6255C331.962 13.545 332.785 16.15 332.785 19.3719V28.7635H336.247V32.7395H325.073V28.7635H328.5V20.0231C328.5 17.5895 328.192 16.8355 327.712 15.91C326.684 14.0934 324.867 13.0308 322.674 13.0308C319.246 13.0308 316.607 15.3273 316.607 20.0917V28.7635H320.069V32.7395H308.895V28.7635H312.322V5.07883H308.895V1.10283H316.572Z"
          fill="#748491"
        />
        <path
          d="M347.592 6.62125C347.592 10.0146 345.159 12.2425 342.074 12.2425C338.989 12.2425 336.555 10.0146 336.555 6.62125C336.555 3.22793 338.989 1 342.074 1C345.159 1 347.592 3.22793 347.592 6.62125ZM337.926 6.62125C337.926 9.2605 339.777 11.0086 342.074 11.0086C344.336 11.0086 346.187 9.2605 346.187 6.62125C346.187 3.98201 344.336 2.23393 342.039 2.23393C339.777 2.19966 337.926 3.98201 337.926 6.62125ZM341.662 3.53642C342.896 3.53642 344.062 3.8449 344.062 5.35304C344.062 6.14139 343.65 6.75835 342.999 7.10111L343.205 7.47815C343.719 8.4036 344.13 9.02057 344.507 9.5347C344.576 9.63753 344.542 9.70608 344.165 9.70608H343.65C343.342 9.70608 343.239 9.70608 343.136 9.56898C342.931 9.32905 342.622 8.84919 342.074 7.82091L341.799 7.30677H341.32V9.56898C341.32 9.67181 341.182 9.70608 340.737 9.70608H340.6C340.12 9.70608 340.051 9.70608 340.051 9.56898V3.67352C340.051 3.57069 340.086 3.53642 340.428 3.53642H341.662ZM341.285 6.31277H341.697C342.108 6.31277 342.382 6.20994 342.554 6.03856C342.725 5.86718 342.793 5.66153 342.793 5.38732C342.793 5.11311 342.725 4.94173 342.588 4.80463C342.416 4.63325 342.108 4.5647 341.662 4.5647H341.285V6.31277Z"
          fill="#748491"
        />
      </g>
    </g>
    <defs>
      <clippath
        id="clip0_2_2304"
      >
        <rect
          fill="white"
          height="41.1866"
          width="347.59"
        />
      </clippath>
      <clippath
        id="clip1_2_2304"
      >
        <rect
          fill="white"
          height="40"
          width="347.592"
        />
      </clippath>
    </defs>
  </svg>
</DocumentFragment>
`;

exports[`should render default size 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="16"
    viewBox="0 0 348 42"
    width="136"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      clip-path="url(#clip0_2_2304)"
    >
      <g
        clip-path="url(#clip1_2_2304)"
      >
        <path
          d="M25.7069 1.06855V26.2614H28.2776V32.8081H18.6804V30.2716C17.138 32.2596 14.9786 33.2879 12.0651 33.2879C5.89546 33.2536 0 27.8723 0 20.6401C0 13.6135 5.75835 8.33505 11.8937 8.33505C14.4987 8.33505 16.7952 9.29477 18.6804 11.0771V7.58098H16.1097V1.06855H25.7069ZM18.886 20.6401C18.886 17.4867 16.3153 14.9846 13.0591 14.9846C9.80291 14.9846 7.30077 17.6924 7.30077 20.8115C7.30077 23.9991 9.97429 26.5698 13.0591 26.5698C16.3839 26.6041 18.886 23.7935 18.886 20.6401Z"
          fill="#EF3200"
        />
        <path
          d="M40.2057 8.84919V26.2614H42.7078V32.8081H30.6084V26.2614H33.1448V15.3273H30.6084V8.84919H40.2057ZM40.2057 7.58098H33.1448V1.06855H40.2057V7.58098Z"
          fill="#EF3200"
        />
        <path
          d="M58.2691 8.74636H62.9306V15.6701H56.5553C56.4182 14.2991 55.5613 13.545 54.156 13.545C52.8535 13.545 51.8937 14.2648 51.8937 15.2931C51.8937 16.6298 53.2991 17.2125 56.7267 18.0351C59.7772 18.7892 61.6967 19.7147 62.862 20.9143C64.0274 22.1482 64.6101 23.7249 64.6101 25.6101C64.6101 29.9974 61.1482 33.2879 55.9383 33.2879C53.3676 33.2879 51.5853 32.7052 49.8372 31.2314V32.7395H45.0043V25.8843H51.7566C52.0309 27.5296 52.9563 28.2494 54.7386 28.2494C56.3496 28.2494 57.138 27.3925 57.138 26.3299C57.138 25.1302 56.2811 24.5133 53.4362 23.7249L51.2082 23.108C49.8372 22.7652 48.7746 22.3539 47.7806 21.7369C45.8269 20.503 44.4216 18.6178 44.4216 15.8758C44.4216 11.3513 48.2262 8.33505 53.1277 8.33505C55.3556 8.33505 56.898 8.81491 58.2691 10.2545V8.74636Z"
          fill="#EF3200"
        />
        <path
          d="M81.988 33.2536C79.7601 33.2536 77.4636 32.7052 74.8929 30.5116V34.5219H77.4293V41H65.2956V34.5219H67.832V15.3273H65.2956V8.84919H74.8929V11.6255C76.4353 9.46615 78.6975 8.33505 81.5424 8.33505C88.0206 8.33505 93.6761 13.8877 93.6761 20.9143C93.6418 27.461 88.3976 33.2536 81.988 33.2536ZM86.4439 20.7772C86.4439 17.4524 83.7361 15.0189 80.4799 15.0189C77.2236 15.0189 74.6187 17.6924 74.6187 20.8458C74.6187 24.2734 77.4636 26.6041 80.5141 26.6041C83.8389 26.6041 86.4439 23.9649 86.4439 20.7772Z"
          fill="#EF3200"
        />
        <path
          d="M113.076 8.84919H122.674V15.3273H120.137V26.2614H122.674V32.8081H113.076V30.7515C111.534 32.5681 109.272 33.5621 106.495 33.5621C99.7429 33.5621 94.4644 27.5981 94.4644 20.9486C94.4644 13.922 100.017 8.09512 106.769 8.09512C109.443 8.09512 111.465 9.08912 113.076 11.1457V8.84919ZM113.111 20.7429C113.111 17.5895 110.608 15.0189 107.386 15.0189C104.165 15.0189 101.628 17.7266 101.628 20.88C101.628 24.1705 104.302 26.6041 107.386 26.6041C110.643 26.6041 113.111 23.8963 113.111 20.7429Z"
          fill="#EF3200"
        />
        <path
          d="M135.116 1.06855V8.84919H138.406V15.3273H135.116V26.2614H138.543V32.7738H128.055V15.3273H124.319V8.84919H128.055V1.06855H135.116Z"
          fill="#EF3200"
        />
        <path
          d="M157.224 8.84919H163.188V17.4524H156.607C155.407 15.9443 154.036 15.0189 152.048 15.0189C148.86 15.0189 146.632 17.7267 146.632 20.8458C146.632 24.102 149.066 26.6041 152.288 26.6041C154.036 26.6041 155.544 25.85 156.367 24.3762H164.182C162.571 29.7918 157.601 33.5279 152.048 33.5279C144.987 33.5279 139.469 27.6324 139.469 20.7772C139.469 13.8535 144.919 8.06084 151.465 8.06084C153.625 8.06084 155.51 8.81491 157.224 10.5973V8.84919Z"
          fill="#EF3200"
        />
        <path
          d="M174.327 1.06855V12.0026C175.938 9.56898 178.098 8.36932 180.908 8.36932C186.187 8.36932 189.443 12.8595 189.443 18.4807V26.2614H191.945V32.8081H182.348V20.1602C182.348 16.6641 181.045 15.2588 178.44 15.2588C175.733 15.2588 174.293 17.0411 174.293 20.0231V26.2271H176.829V32.7738H164.696V26.2614H167.232V7.58098H164.696V1.06855H174.327Z"
          fill="#EF3200"
        />
        <path
          d="M203.256 1.10283V13.2365C205.039 10.3573 207.369 8.74636 210.626 8.74636C213.093 8.74636 215.356 9.70608 217.001 11.6255C218.646 13.545 219.469 16.15 219.469 19.3719V28.7635H222.931V32.7395H211.757V28.7635H215.184V20.0231C215.184 17.5895 214.876 16.8355 214.396 15.91C213.368 14.0934 211.551 13.0308 209.357 13.0308C205.93 13.0308 203.29 15.3273 203.29 20.0917V28.7635H206.752V32.7395H195.578V28.7635H199.006V5.07883H195.578V1.10283H203.256Z"
          fill="#748491"
        />
        <path
          d="M247.404 22.5938H227.318C228.175 26.8783 231.397 29.5518 235.476 29.5518C238.355 29.5518 240.48 28.3865 242.125 25.9529H246.615C244.456 30.7515 240.24 33.3907 235.476 33.3907C228.723 33.3907 223.102 27.9752 223.102 21.0514C223.102 13.9563 228.689 8.60925 235.304 8.60925C241.851 8.60925 247.541 13.7164 247.541 21.12C247.541 21.6684 247.472 22.2853 247.404 22.5938ZM243.599 19.2691C242.674 15.156 239.314 12.4482 235.27 12.4482C231.74 12.4482 228.278 15.0189 227.318 19.2691H243.599Z"
          fill="#748491"
        />
        <path
          d="M270.231 9.2605H277.943V13.2365H274.584V28.7635H277.943V32.7395H270.231V28.7635C267.935 31.9512 265.261 33.2536 261.731 33.2536C255.081 33.2536 249.803 27.7009 249.803 21.1542C249.803 14.539 255.116 8.78063 261.902 8.78063C265.433 8.78063 268.003 10.1517 270.231 13.3393V9.2605ZM270.231 20.9143C270.231 16.6298 266.735 13.0308 262.108 13.0308C257.618 13.0308 254.122 16.6984 254.122 21.0514C254.122 25.4387 257.652 29.0034 262.108 29.0034C266.564 28.9692 270.231 25.4387 270.231 20.9143Z"
          fill="#748491"
        />
        <path
          d="M289.049 1.10283V28.7635H292.476V32.7395H281.097V28.7635H284.73V5.07883H281.268V1.10283H289.049Z"
          fill="#748491"
        />
        <path
          d="M300.6 1.10283V9.2605H305.124V13.2365H300.6V28.7635H305.433V32.7395H296.315V13.2365H292.236V9.2605H296.315V1.10283H300.6Z"
          fill="#748491"
        />
        <path
          d="M316.572 1.10283V13.2365C318.355 10.3573 320.685 8.74636 323.942 8.74636C326.41 8.74636 328.672 9.70608 330.317 11.6255C331.962 13.545 332.785 16.15 332.785 19.3719V28.7635H336.247V32.7395H325.073V28.7635H328.5V20.0231C328.5 17.5895 328.192 16.8355 327.712 15.91C326.684 14.0934 324.867 13.0308 322.674 13.0308C319.246 13.0308 316.607 15.3273 316.607 20.0917V28.7635H320.069V32.7395H308.895V28.7635H312.322V5.07883H308.895V1.10283H316.572Z"
          fill="#748491"
        />
        <path
          d="M347.592 6.62125C347.592 10.0146 345.159 12.2425 342.074 12.2425C338.989 12.2425 336.555 10.0146 336.555 6.62125C336.555 3.22793 338.989 1 342.074 1C345.159 1 347.592 3.22793 347.592 6.62125ZM337.926 6.62125C337.926 9.2605 339.777 11.0086 342.074 11.0086C344.336 11.0086 346.187 9.2605 346.187 6.62125C346.187 3.98201 344.336 2.23393 342.039 2.23393C339.777 2.19966 337.926 3.98201 337.926 6.62125ZM341.662 3.53642C342.896 3.53642 344.062 3.8449 344.062 5.35304C344.062 6.14139 343.65 6.75835 342.999 7.10111L343.205 7.47815C343.719 8.4036 344.13 9.02057 344.507 9.5347C344.576 9.63753 344.542 9.70608 344.165 9.70608H343.65C343.342 9.70608 343.239 9.70608 343.136 9.56898C342.931 9.32905 342.622 8.84919 342.074 7.82091L341.799 7.30677H341.32V9.56898C341.32 9.67181 341.182 9.70608 340.737 9.70608H340.6C340.12 9.70608 340.051 9.70608 340.051 9.56898V3.67352C340.051 3.57069 340.086 3.53642 340.428 3.53642H341.662ZM341.285 6.31277H341.697C342.108 6.31277 342.382 6.20994 342.554 6.03856C342.725 5.86718 342.793 5.66153 342.793 5.38732C342.793 5.11311 342.725 4.94173 342.588 4.80463C342.416 4.63325 342.108 4.5647 341.662 4.5647H341.285V6.31277Z"
          fill="#748491"
        />
      </g>
    </g>
    <defs>
      <clippath
        id="clip0_2_2304"
      >
        <rect
          fill="white"
          height="41.1866"
          width="347.59"
        />
      </clippath>
      <clippath
        id="clip1_2_2304"
      >
        <rect
          fill="white"
          height="40"
          width="347.592"
        />
      </clippath>
    </defs>
  </svg>
</DocumentFragment>
`;
